### PR TITLE
Enable tags and disable random branch artifacts

### PR DESCRIPTION
**Pull request by oguzkocer**
This PR updates `publish-to-s3-gradle-plugin` to `0.2.2` which includes [this fix](https://github.com/Automattic/publish-to-s3-gradle-plugin/pull/4). It also utilizes the `should-publish-library-artifacts` command from [this PR](https://github.com/wordpress-mobile/circleci-orbs/pull/73). @jkmassel Since `only build pull requests` option didn't do exactly what we wanted, I disabled that and introduced this command instead. Let me know what you think!

This PR is also supposed to enable publishing artifacts from all tags, but I couldn't get it to work. @jkmassel @mokagio Maybe you can help me with that?

This is probably the last PR before we merge everything to `develop` and update the clients. I am also planning to post an announcement for it. Let me know if I am missing anything!

### CODE REVIEW

**Review by mokagio — CHANGES_REQUESTED** [review]
I triggered a new build of this PR because I was confused by the fact that its first run [did not publish artifacts](https://app.circleci.com/pipelines/github/wordpress-mobile/WordPress-Utils-Android/304/workflows/e66a8144-89ba-46fb-8178-2df7c792eba4/jobs/666):

image

@oguzkocer: I'm pretty sure the reason would you pushed the branch first then opened the PR. Is that right? That would have resulted in CircleCI starting a build (because we don't have "only build PRs anymore") that did have the tag and PR environment variable null, and the branch different from `develop` or `trunk`, which made the workflow halt, _as desired_.

I verified this behavior (but not using this branch) by:

1. Pushing a branch and waiting for its build to start. [It did not publish to S3](https://app.circleci.com/pipelines/github/wordpress-mobile/WordPress-Utils-Android/307/workflows/8ca46a8d-b84d-4669-a9a0-0b516aab68b1/jobs/672).
2. Opening a PR and restarting the workflow for that same commit. [It did publish to S3](https://app.circleci.com/pipelines/github/wordpress-mobile/WordPress-Utils-Android/307/workflows/cee33080-86e2-4ec3-9e65-1a0f470bf4da/jobs/673)

---

> This PR is also supposed to enable publishing artifacts from all tags, but I couldn't get it to work. @jkmassel @mokagio Maybe you can help me with that?

As far as I could see, the issue is that the publish job depends on the test job, which is not configured to run on tags. One would expect CircleCI to work this out automatically, but that doesn't seem to be the case 🙊 .

I [added the filter to the `Test` job](https://github.com/wordpress-mobile/WordPress-Utils-Android/commit/3f00e5f0a6a8b689f82dbdb0862c2d8f9e4fa651) and verified a tag triggered the workflow in CI:

image

Requesting changes to track the need to implement (cherry-pick?) that extra configuration.

**Comment by oguzkocer** [comment]
@mokagio Thanks a lot for the handling the tagging issue and opening that orb improvement PR 🙇 This is ready for another look!

**Review by mokagio — APPROVED** [review]
I can see CI successfully published to S3 from this branch, which validates the rename on the orb side as well as the rest of the pipeline logic.

image

I pushed a test tag and verified [CI worked in that scenario](https://app.circleci.com/pipelines/github/wordpress-mobile/WordPress-Utils-Android/313/workflows/a72675ff-0209-4510-bdf9-77c304594097), too.

image

File: .circleci/config.yml
Comment (mokagio) [inline .circleci/config.yml]: Up to you whether to wait for https://github.com/wordpress-mobile/circleci-orbs/pull/73 to be merged and update this before merging it, or point to the main orb in the parent PR  https://github.com/wordpress-mobile/WordPress-Utils-Android/pull/63

Comment (oguzkocer) [reply .circleci/config.yml]: I've updated this in 2852c5f9567e17cb9253e6682b95af6a6526eb8e. I'll merge it once CI is green 🤞

**Pull request merged**
